/* defines protocol buffers sent to clients */
syntax = "proto3";
package sadfs.proto.client;
import "common.proto";

// response for a chunk_location_request
message chunk_location_response {
	bool            ok          = 1;
	repeated string server_ips  = 2;
	repeated int32  ports       = 3;  // not uint32 for sadfs code compatibility
	bytes           chunk_id    = 4;
	int32           version_num = 5;
	uint32		file_size   = 6;
}

// response for a file_metadata_request
message file_metadata_response {
	bool		ok	    = 1;
	uint32		size	    = 2;
}

// response for a chunk read request
message read_response {
	bool  ok   = 1;
	bytes data = 2;
}

// container for all control messages sent to clients
message message_container {
	sadfs.proto.header  header = 1;
	oneof               msg {
		sadfs.proto.acknowledgement ack                = 6;
		chunk_location_response     chunk_location_res = 7;
		file_metadata_response      file_metadata_res  = 8;
		read_response               read_res           = 9;
	}
}
